### PR TITLE
8276490: Incorrect path for duplicate x and y values, when path falls outside axis bound

### PR DESCRIPTION
**Pull request by abhinayagarwal**
PathElements were skipped in AreaChart if the data point were outside axis bounds and had duplicate value for either x or y. This is now fixed with this PR.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8276490](https://bugs.openjdk.java.net/browse/JDK-8276490): Incorrect path for duplicate x and y values, when path falls outside axis bound

### Reviewers
 * [Ajit Ghaisas](https://openjdk.java.net/census#aghaisas) (@aghaisas - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/667/head:pull/667` \
`$ git checkout pull/667`

Update a local copy of the PR: \
`$ git checkout pull/667` \
`$ git pull https://git.openjdk.java.net/jfx pull/667/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 667`

View PR using the GUI difftool: \
`$ git pr show -t 667`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/667.diff">https://git.openjdk.java.net/jfx/pull/667.diff</a>

</details>

### CODE REVIEW

File: modules/javafx.controls/src/test/java/test/javafx/scene/chart/AreaChartTest.java [hunk 1]
```diff
@@ -151,4 +154,58 @@ public void testSeriesRemoveAnimatedStyleClasses() {
...
+
+        final NumberAxis xAxis = (NumberAxis) ac.getXAxis();
+        final NumberAxis yAxis = (NumberAxis) ac.getYAxis();
+
+        LineTo lastPath = findLastLineInPath();
+        assertEquals(100d, xAxis.getValueForDisplay(lastPath.getX()).doubleValue(), 0.001);
+        assertEquals(20d, yAxis.getValueForDisplay(lastPath.getY()).doubleValue(), 0.001);
+    }
+
+    @Test public void testPathOutsideXBoundsWithDuplicateXAndLowerY() {
```
[the remark this answers is not in the document]
Comment (abhinayagarwal) [reply modules/javafx.controls/src/test/java/test/javafx/scene/chart/AreaChartTest.java:193]: I have added a couple of additional tests

**Comment by aghaisas** [comment]
Looks good to me.

**Review by aghaisas — APPROVED** [review]

**Pull request closed without merging**